panacea/web/static/js/app.js

```diff
@@ -27,35 +27,49 @@ async function submitFile() {
     if (drugsResponse.ok) {
       const data = await drugsResponse.json();
       const drugs = data.drugs;
-      displayDrugs(drugs);
+      if (drugs.length > 0){
+        displayDrugs(drugs);
 
-      const urisResponse = await fetch("/api/uris", {
-        method: 'POST',
-        body: JSON.stringify({labels: drugs}),
-        credentials: 'same-origin',
-        headers: defaultHeaders
-      });
-
-      if (urisResponse.ok) {
-        const data = await urisResponse.json();
-        const uris = Object.keys(data.uris.found);
-
-        const ddisResponse = await fetch("/api/ddis", {
+        const urisResponse = await fetch("/api/uris", {
           method: 'POST',
-          body: JSON.stringify({drugs: uris}),
+          body: JSON.stringify({labels: drugs}),
           credentials: 'same-origin',
           headers: defaultHeaders
         });
 
-        if (ddisResponse.ok) {
-          const data = await ddisResponse.json();
-          const ddis = data.ddis;
-          displayDdis(ddis);
+        if (urisResponse.ok) {
+          const data = await urisResponse.json();
+          const uris = Object.keys(data.uris.found);
+          const unidentifiedDrugs = Object.values(data.uris.not_found);
+          if (unidentifiedDrugs.length > 0){
+            displayUnidentifiedDrugs(unidentifiedDrugs);
+          }
+          if (uris.length >= 2){
+            const ddisResponse = await fetch("/api/ddis", {
+              method: 'POST',
+              body: JSON.stringify({drugs: uris}),
+              credentials: 'same-origin',
+              headers: defaultHeaders
+            });
+
+            if (ddisResponse.ok) {
+              const data = await ddisResponse.json();
+              const ddis = data.ddis;
+              displayDdis(ddis);
+            } else {
+              displayError(await ddisResponse.json());
+            }
+          }
+          else {
+            // TODO: ddis require more than one drug by definition
+            console.log("less than two identified drugs, no ddis")
+          }
         } else {
-        displayError(await ddisResponse.json());
+          displayError(await urisResponse.json());
         }
-      } else {
-        displayError(await urisResponse.json());
+      }
+      else {
+        displayError({"message": "No drugs found"});
```

Comment:
this is old syntax, `{message: "bleugh"}` would be nicer, i see we have used it above